ingestion_tools/scripts/data_validation/tests/test_tiltseries.py

```diff
@@ -0,0 +1,122 @@
+
+        def check_size_in_mrc(header, _interpreter, _mrc_filename, tiltseries_metadata):
+            del _interpreter, _mrc_filename
+            assert header.nx == tiltseries_metadata["size"]["x"]
+            assert header.ny == tiltseries_metadata["size"]["y"]
+            assert header.nz == tiltseries_metadata["size"]["z"]
+
+        self.mrc_header_helper(check_size_in_mrc, tiltseries_metadata=tiltseries_metadata)
+
+    def test_tiltseries_scales_in_zarr(self, tiltseries_metadata: Dict):
+        """Check that the tiltseries binning factor scales matches the Zarr header volume size."""
+
+        def check_size_in_zarr(header_data, _zarr_filename, tiltseries_metadata):
+            del _zarr_filename
+            for binning_factor in BINNING_FACTORS:  # Check all binning factors
+                assert header_data["zarrays"][binning_factor]["shape"] == [
+                    tiltseries_metadata["scales"][binning_factor]["z"],
+                    tiltseries_metadata["scales"][binning_factor]["y"],
+                    tiltseries_metadata["scales"][binning_factor]["x"],
+                ]
+
+        self.zarr_header_helper(check_size_in_zarr, tiltseries_metadata=tiltseries_metadata)
+
+    def test_tiltseries_zarr_matches(
+        self,
+        tiltseries_metadata: Dict,
+        tiltseries_zarr_header: Dict[str, Dict[str, Dict]],
+    ):
+        """Check that the metadata-listed zarr file matches the actual file."""
+        assert len(tiltseries_zarr_header) == 1
+        assert tiltseries_metadata["omezarr_dir"] == os.path.basename(list(tiltseries_zarr_header.keys())[0])
+
+    def test_tiltseries_mrc_matches(
+        self,
+        tiltseries_metadata: Dict,
+        tiltseries_mrc_header: Dict[str, MrcInterpreter],
+    ):
+        """Check that the metadata-listed mrc file matches the actual file."""
+        assert len(tiltseries_metadata["mrc_files"]) == 1
+        assert len(tiltseries_mrc_header) == 1
+        assert tiltseries_metadata["mrc_files"][0] == os.path.basename(list(tiltseries_mrc_header.keys())[0])
+
+    ### END metadata-MRC/Zarr consistency tests ###
+
+    ### BEGIN MDOC consistency tests ###
+    def test_tiltseries_pixel_spacing_mdoc(self, tiltseries_metadata: Dict, tiltseries_mdoc: pd.DataFrame):
+        """Check that the tiltseries pixel spacing matches the MDOC data."""
+        assert len(set(tiltseries_mdoc["PixelSpacing"])) == 1
+        assert tiltseries_metadata["pixel_spacing"] == tiltseries_mdoc["PixelSpacing"][0]
+
+    def test_tiltseries_image_size_mdoc(self, tiltseries_metadata: Dict, tiltseries_mdoc: pd.DataFrame):
+        """Check that the tiltseries image size matches the MDOC data."""
+        assert len(set(tiltseries_mdoc["ImageSize"])) == 1
+        assert tiltseries_metadata["size"]["x"] == tiltseries_mdoc["ImageSize"][0][0]
+        assert tiltseries_metadata["size"]["y"] == tiltseries_mdoc["ImageSize"][0][1]
+
+    ### END MDOC consistency tests ###
```

Comment:
While the mdoc files are saved along with the tiltseries at the moment, they are associated to the frames. As, there is processing to go from frames to tiltseries, it is very unlikely for us to have the same Image size or pixel spacings. So, these two tests are more likely to fail and that is the expected behaviour.

Reply:
Would it be of interest to see which runs fail against these tests though? Maybe worth asking Utz tomorrow?